src/ast/query.rs

```diff
@@ -1865,27 +1873,57 @@ impl fmt::Display for SelectInto {
     }
 }
 
+#[derive(Debug, Copy, Clone, PartialEq, PartialOrd, Eq, Ord, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "visitor", derive(Visit, VisitMut))]
+pub enum WithModifier {
```

Comment:
Can we add a description and link to the clickhouse docs here? currently its only specified in the `ALL` variant which seems uneven in terms of placement.
Also it would help in this case with a syntax example in the comment like `GROUP BY year WITH ROLLUP` so its clear to folks at a glance what it does without consulting the docs